types/mempool/sender_nonce.go

```diff
@@ -0,0 +1,201 @@
+package mempool
+
+import (
+	crand "crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"math/rand"
+
+	huandu "github.com/huandu/skiplist"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+)
+
+var (
+	_ Mempool  = (*senderNonceMempool)(nil)
+	_ Iterator = (*senderNonceMepoolIterator)(nil)
+)
+
+type senderTxs struct {
```

Comment:
Can we simplify usages of `senderTxs` to just `cursor *huandu.Element`?  The wrapping type seems to serve no purpose.